File: pytools/examples/golang/src/server/proto/req_test_php.go
package proto

import (
	"packet"
)

type ReqTestPhp struct {
	u64                      uint64
	strxx                    string
	msgReq                   *MsgTestPhp
	msgOptFlag               uint8
	msgOpt                   *MsgTestPhp
	msgRep                   []*MsgTestPhp
}

func ReqTestPhpDecode(pack *packet.Packet) *ReqTestPhp {
	reqTestPhp := &ReqTestPhp{}

	reqTestPhp.u64 = pack.ReadUint64()
	reqTestPhp.strxx = pack.ReadString()
	reqTestPhp.msgReq = MsgTestPhpDecode(pack)
	reqTestPhp.msgOptFlag = pack.ReadUint8()
	if reqTestPhp.msgOptFlag == 1 {
		reqTestPhp.msgOpt = MsgTestPhpDecode(pack)
	}
	msgRepCount := pack.ReadUint16()
	for ;msgRepCount > 0; msgRepCount-- {
		reqTestPhp.msgRep = append(reqTestPhp.msgRep, MsgTestPhpDecode(pack))
	}
	return reqTestPhp
}

func (this *ReqTestPhp) GetU64() uint64 {
	return this.u64
}

func (this *ReqTestPhp) GetStrxx() string {
	return this.strxx
}

func (this *ReqTestPhp) GetMsgReq() *MsgTestPhp {
	return this.msgReq
}

func (this *ReqTestPhp) GetMsgOpt() *MsgTestPhp {
	return this.msgOpt
}

func (this *ReqTestPhp) GetMsgRep() []*MsgTestPhp {
	return this.msgRep
}

File: pytools/examples/java/server/protocol/ReqTestPhp.java
package protocol;


import network.Packet;
import network.PacketUtil;

import java.util.List;
import java.util.ArrayList;


public class ReqTestPhp
{
	private long u64;
	private String strxx;
	private MsgTestPhp msg_req;
	private byte msg_opt_flag;
	private MsgTestPhp msg_opt;
	private List<Msgtestphp> msg_rep = new ArrayList<Msgtestphp>();


	public ReqTestPhp(Packet packet)
	{
		u64 = packet.readLong();
		strxx = packet.readString();
		msg_req = new MsgTestPhp(packet);
		msg_opt_flag = packet.readByte();
		if (msg_opt_flag == 1)
		{
			msg_opt = new MsgTestPhp(packet);
		}
		int msg_rep_count = PacketUtil.readUShort(packet.readShort());
		for (int i = 0; i < msg_rep_count; i++)
		{
			msg_rep.add(new MsgTestPhp(packet));
		}
	}



	public long getU64()
	{
		return this.u64;
	}

	public String getStrxx()
	{
		return this.strxx;
	}

	public MsgTestPhp getMsgReq()
	{
		return this.msg_req;
	}

	public MsgTestPhp getMsgOpt()
	{
		return this.msg_opt;
	}

	public List<Msgtestphp> getMsgRep()
	{
		return this.msg_rep;
	}

}
